pkg/engine/systemd.go

```diff
@@ -63,22 +63,24 @@ func enableRestartSystemdService(mo *SingleMethodObj, action, dest, service stri
 	}
 
 	// TODO: remove
-	os.Setenv("ROOT", "true")
-	if !sd.Root {
-		//os.Setenv("ROOT", "false")
-		klog.Info("At this time, fetchit non-root user cannot enable systemd service on the host")
-		klog.Infof("To enable this non-root service, run 'systemctl --user enable %s' on host machine", service)
-		klog.Info("To enable service as root, run with Systemd.Root = true")
-		return nil
+	if sd.Root {
+		os.Setenv("ROOT", "true")
+	} else {
+		os.Setenv("ROOT", "false")
 	}
-
 	s := specgen.NewSpecGenerator(systemdImage, false)
 	runMounttmp := "/run"
 	runMountsd := "/run/systemd"
 	runMountc := "/sys/fs/cgroup"
+	xdg := ""
 	if !sd.Root {
-		runMountsd = "/run/user/1000/systemd"
-		s.User = "1000"
+		// need to document this for non-root usage
+		// can't use user.Current because always root in fetchit container
+		xdg = os.Getenv("XDG_RUNTIME_DIR")
+		if xdg == "" {
+			xdg = "/run/user/1000"
```

Comment:
Should we add in `-e USER=%U` to the systemd start then we can use os.Getenv("USER") to set the value rather than 1000?

Although this passes CI which has a userid of 1001 it may not be actually necessary

Reply:
Either would work, but only 1 is necessary (the XDG_RUNTIME_DIR _or_ USER) but I think XDG_RUNTIME_DIR is less confusing because it won't be confused with passing `--user`. We'll want to document clearly, and this is only necessary if non-root podman user != 1000. I think it's reasonable to default to 1000 since that's the default UID of a non-root user, at least w/ RH-based OS. Passing the -e USER will be confusing, since the sysd pod, I _think_ has to run w/ privileged - (unless it doesn't?) - not w/ `sudo podman` but with root _inside_ pod.

Reply:
I think 1000 is just the first user added to a system. I mean I couldn't see a lot of users being added to a system

Reply:
oh im dumb i just looked at the systemd entry...../me runs away

`-e XDG_RUNTIME_DIR="/run/user/%U"`